### ISSUE TITLE
Ionic 2 RC2 Polyfill.js Argument not optional on Windows 10 mobile

### ISSUE THREAD

**Issue by mtycholaz** [driftyco/ionic#9051]
Updated to the latest RC2, but I now get errors when deploying to the emulator/device.

Here's what I've tried

1. `ionic start test --v2`
2. `ionic platform add windows`
3. `npm run build --dev`
4. Open up `CordovaApp.sln`
5. Set the device to mobile emulator and deploy
![image](https://cloud.githubusercontent.com/assets/3585379/20032893/8b8836da-a359-11e6-8c5d-cdec3e95ad04.png)

And this is the error that happens in the Polyfill.js file Exception about `Argument not optional`

![image](https://cloud.githubusercontent.com/assets/3585379/20032852/ea929f5e-a358-11e6-804f-6fc24d2f57cf.png)

Here's what the package.json file has, but it's a vanilla project soo...

```
{
  "name": "ionic-hello-world",
  "author": "Ionic Framework",
  "homepage": "http://ionicframework.com/",
  "private": true,
  "scripts": {
    "build": "ionic-app-scripts build",
    "watch": "ionic-app-scripts watch",
    "serve:before": "watch",
    "emulate:before": "build",
    "deploy:before": "build",
    "build:before": "build",
    "run:before": "build"
  },
  "dependencies": {
    "@angular/common": "2.1.1",
    "@angular/compiler": "2.1.1",
    "@angular/compiler-cli": "2.1.1",
    "@angular/core": "2.1.1",
    "@angular/forms": "2.1.1",
    "@angular/http": "2.1.1",
    "@angular/platform-browser": "2.1.1",
    "@angular/platform-browser-dynamic": "2.1.1",
    "@angular/platform-server": "2.1.1",
    "@ionic/storage": "1.1.6",
    "ionic-angular": "2.0.0-rc.2",
    "ionic-native": "2.2.3",
    "ionicons": "3.0.0",
    "rxjs": "5.0.0-beta.12",
    "zone.js": "0.6.21"
  },
  "devDependencies": {
    "@ionic/app-scripts": "0.0.39",
    "typescript": "2.0.6"
  },
  "description": "test: An Ionic project",
  "cordovaPlugins": [
    "cordova-plugin-device",
    "cordova-plugin-console",
    "cordova-plugin-whitelist",
    "cordova-plugin-splashscreen",
    "cordova-plugin-statusbar",
    "ionic-plugin-keyboard"
  ],
  "cordovaPlatforms": []
}

```

I've even stripped out everything to this, and it still get the error.

```
<!DOCTYPE html>
<html lang="en" dir="ltr">
<head>
</head>
<body class="platform-windows platform-cordova platform-webview">
  <script src="build/polyfills.js"></script>
</body>
</html>
```

**Comment by jagansai** [comment]
ionic serve also gives this error in RC2. The application was working fine in RC1 and when I update it to RC2, I get this error. No changes made to the code.

**Comment by morious** [comment]
I am seeing the same thing...followed by:
```
Unable to get property '__symbol__' of undefined or null reference
in main.js
```
Windows 10, Edge.

**Comment by mtycholaz** [comment]
Yes, I have this too. Has to do with the Zones.js somehow? Workaround is currently to revert to RC 1

**Comment by jagansai** [comment]
Don't revert to RC1. It works fine in Chrome. Looks like an issue with Edge or non-standard behavior tailored for chrome ( which is the case these days ). If you need to test it for Windows Phone then no other option but to revert to RC1.

**Comment by mtycholaz** [comment]
Yah, I don't have any other option, but reverting. I utilize WinRT API's that don't exist in Chrome, or have Cordova plugins. My workflow is make a code change, compile with cli, build in Visual Studio, and then deploy to either a device or emulator. 

Until Windows 10 UWP gets more attention from developers, this is just how it is.

**Comment by mpaland** [comment]
Related to #9025

**Comment by sergeyrolich** [comment]
I have this problem too. You have any ideas?

**Comment by mtycholaz** [comment]
It's definitely a problem in Edge. Even viewing the online documentation doesn't work with the side preview. Either develop in chrome until it's fixed, or revert.

**Comment by infinnie** [comment]
Which Edge version? Edge 13 seems to work fine.

**Comment by infinnie** [comment]
Edge 14? Seems that the problem has something to do with `window.fetch()`.

**Comment by mtycholaz** [comment]
I'm running Edge 14. RC1 didn't have this issue with Edge 14, so it was something changed in Polyfill.js

**Comment by infinnie** [comment]
@mtycholaz See my second comment.

**Comment by mtycholaz** [comment]
Interesting, I'll try that.

**Comment by jgw96** [comment]
Hello all! I am looking into this issue today. It seems that edge's implementation of the `fetch` api is different than other browsers (chrome, firefox, opera) which is causing an issue with the polyfills.

**Comment by jgw96** [comment]
Is anyone here running a nightly build of edge? If so does this issue still happen in that nightly build?

**Comment by mtycholaz** [comment]
I can confirm Edge 15 on Windows 10 mobile has this issue, along with Edge 14 anniversary edition.

**Comment by mtycholaz** [comment]
However, in Edge 13 everything works.

**Comment by mpaland** [comment]
Nope, just normal, actual Edge 14.14393. Doesn't work!

**Comment by jagansai** [comment]
I  am not sure Edge has nightly build. If there is, please help me with the link. I am running Windows Insider Build 14965 ( latest ). It still does not work in the Edge version that comes with that build.

**Comment by mtycholaz** [comment]
Edge 13.10586 (Threshold) - Works
Edge 14.14393 Desktop (Anniversary edition) - Error
Edge 14.14393 Mobile (Anniversary edition) - Error
Edge 15.14955 Mobile (Insider build) - Error

**Comment by vaskou** [comment]
@jgw96 
I did some digging and I found something about this problem.
Ms Edge and Chrome throws an exception at polyfill.js, line 3, col 16552. Chrome continues, but Ms Edge stops. 
The code that has the problem is var N=t.fetch();

I compared the polyfill.js of RC1 (which worked) with the one of RC2. 
In RC1 the piece of code (N=t.fetch();) is in a try - catch, but in RC2 is not.

Then I run gulp polyfill so I can get the polyfill.js of the latest release (RC3) of the ionic repository.
The polyfill.js file that it produced(in dist/ionic-angular/polyfills direcory), was the same as RC1 file (with the try - catch). So I think there is some mistake in the distribution that we get from the npm.

Sorry for my English. I you have any questions please ask me.

**Comment by jgw96** [comment]
Hello everyone! We have updated to the latest Angular as of this morning (which means it will be in the next release) which includes a fix for this issue. Thanks!

**Comment by mtycholaz** [comment]
Any ETA on the next release?

**Comment by mpaland** [comment]
I've tested the conference app with RC3 and angular 2.2.1 - and as far as I can see, it still has the problem above. 
@jgw96 have you tested this with Edge 14 browser?

**Comment by mtycholaz** [comment]
@mpaland I think we need to wait until RC4. @jqe96 made his comment after RC3 was released.

**Comment by mpaland** [comment]
@mtycholaz You don't need to wait for RC4. You can always use the latest branch to test actual changes.
As @jgw96 pointed out, an update of angular (to 2.2.1 as in latest) should fix the bug. But I can't confirm this.

**Comment by mtycholaz** [comment]
@mpaland I'll try that once I get home.

**Comment by mtycholaz** [comment]
@mpaland Just tried updating the packages.json file to Angular 2.2.1. Everything compiles, but the polyfill.js still throws an error. We'll just need to be patient. I tried this in Edge Mobile 14.14393

**Comment by DeusProx** [comment]
Hey! I got the same problem! I tried all combination with ng2.2.1, ng2.1.1, ion.rc1 and ion.rc3 without success. Using ion.rc3 with ng2.2.1 lead to #9015 so i couldn't even build. 
Is there already some date or estimation for rc4? just saw that there are still 12 issues still open.

Also Offtopic: I'm a linux guy and really hate the fact that I have to use windows+vs2015 and need to reload the project in vs2015 after each build and being still not able to deploy the app to my lumia 540 with vs2015.
So I searched for alternatives and found the command `ionic run windows --device -- --arch=ARM --phone` to run from cmd and was able to deploy to my device. Despite the polyfills problems described earlier the app can be started. The loginscreen shows and i can put in my credentials but i can't use my login button. Is there a way to add debugging mode to the cmd command like with android or do I have to depend on vs2015?

**Comment by mtycholaz** [comment]
@DeusProx Are you coding in linux, and then deploying to a device via a Windows VM?

**Comment by jagansai** [comment]
@DeusProx If you are using VS2015, then you can always debug the app using Visual Studio debugger. In your tsconfig.json file if you declared --sourceMap flag, then you should debug the code. That's much better than debugging from the command prompt.

**Comment by DeusProx** [comment]
@jagansai I would like to not even have to start vs2015. That's why i asked for the right commands
@mtycholaz Coding on linux for android and then trying to port it on windows deploying it directly on a lumia 540 with win10.

The problem with the polyfill.js only appears to happen when I try to emulate a w10 phone(which I won't do in future because it uses Hyper-V and I need VT-x for my virtualbox....).
ATM I deploy to the lumia 540 from command line with `ionic run windows -l -c -s --device -- --arch=ARM --phone`. This works fine as long as I don't close the ionic server which is quite logical. When I use it without the live reload server (`ionic run windows --device -- --arch=ARM --phone`) it starts but pushing the login button doesn't do anything and i have no debugging on clientside and can't check the backend as I don't have access... Sry that this is all offtopic. maybe I should create a new issue for this.

**Comment by jagansai** [comment]
@DuesProx cool. Good luck with the restrictions you imposed onto yourself 😊

**Comment by DeusProx** [comment]
@jagansai restrictions? i can't even start my app with vs2015.... it's just annoying to use it

**Comment by mtycholaz** [comment]
@DeusProx Not sure if it's possible without the visual studio remote debugger. But that tool is (currently) only available on Windows. Microsoft just announced VS for Mac. It's not feature complete yet, but it may work on Linux?

**Comment by DeusProx** [comment]
@mtycholaz Sry I probably confused you. I got app debugging `ionic run windows -l -c -s --device -- --arch=ARM --phone` over windows cmd. Afaik the app is served by a live server on the local host to the smart phone. Through this the polyfills problems of this issue doesn't seems to appear & I have system.out.println and stack traces on the cmd. Without the flags `-l -s -c` everything is packed into the apk and deployed to the phone. This makes the app start but doesn't work correctly (& no debugging). 

Deploying over vs2015 I have the same polyfills bug.

**Comment by jgw96** [comment]
Hello all! Is this still an issue with RC3? The version of Angular we are now running includes fixes for the polyfills issue.

**Comment by mtycholaz** [comment]
@jgw96 What version of Angular are you running?

**Comment by jgw96** [comment]
@mtycholaz sorry, mis-spoke on that. So with the latest nightly release we are now running on 2.2.1. To update to that nightly you can change your package.json to look like [this](https://github.com/driftyco/ionic-conference-app/blob/master/package.json).

**Comment by mtycholaz** [comment]
@jgw96 I believe I tired this already and it didn't work, but I'll try again later tonight.

**Comment by mtycholaz** [comment]
@jgw96 Confirmed, I've updated my package.json to the one you provided. Everything works now. Thanks!

**Comment by mpaland** [comment]
Confirmed! Working in nightly. Nice work!

**Comment by almost92** [comment]
tried ngc 2.2.x and ngc 2.1.x with  "ionic-angular": "2.0.0-rc.3-201612081620", on edge 13.10586
and I am still experiencing the issue. 
any ideas ?
